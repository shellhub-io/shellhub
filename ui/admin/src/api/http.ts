import { AxiosInstance } from "axios";
import { setupInterceptorsTo } from "./interceptors";
import * as axiosTs from "./client";
import { Configuration } from "./client";
import { BaseAPI } from "./client/base";

const configuration = new Configuration();
configuration.basePath = `${window.location.protocol}//${window.location.host}`;
configuration.accessToken = localStorage.getItem("cloud_token") || "";

// eslint-disable-next-line import/no-mutable-exports
let adminApi = new axiosTs.AdminApi(configuration);
// eslint-disable-next-line import/no-mutable-exports
let cloudApi = new axiosTs.CloudApi(configuration);

export const createNewClient = () => {
  const newConfiguration = new Configuration();
  newConfiguration.basePath = `${window.location.protocol}//${window.location.host}`;
  newConfiguration.accessToken = localStorage.getItem("cloud_token") || "";
  adminApi = new axiosTs.AdminApi(newConfiguration);
  cloudApi = new axiosTs.CloudApi(newConfiguration);
  return { adminApi, cloudApi };
};

declare module "./client/base" {
  interface BaseAPI {
    getAxios(): AxiosInstance;
    getConfiguration(): Configuration | undefined;
    setConfiguration(configuration: Configuration): void;
  }
}

/** Returns the axios instance */
BaseAPI.prototype.getAxios = function getAxios(this: BaseAPI): AxiosInstance {
  return this.axios;
};

/** Returns the configuration */
// eslint-disable-next-line vue/max-len
BaseAPI.prototype.getConfiguration = function getConfiguration(this: BaseAPI): Configuration | undefined {
  return this.configuration;
};

/** Sets the configuration */
// eslint-disable-next-line vue/max-len
BaseAPI.prototype.setConfiguration = function setConfiguration(this: BaseAPI, configuration: Configuration): void {
  this.configuration = configuration;
};

// eslint-disable-next-line @typescript-eslint/ban-ts-comment
// @ts-ignore
setupInterceptorsTo(adminApi.axios);

export { adminApi, cloudApi };
