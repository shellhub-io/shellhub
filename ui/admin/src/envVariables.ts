/* eslint-disable import/prefer-default-export */
export const envVariables = {
  announcementsEnable: process.env.SHELLHUB_ANNOUNCEMENTS === "true",
  tinyMceKey: process.env.TINYMCE_API_KEY,
};
