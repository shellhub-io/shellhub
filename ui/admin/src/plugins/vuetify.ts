// Styles
// import "@mdi/font/css/materialdesignicons.css";
import "vuetify/styles";

// Vuetify
import { createVuetify } from "vuetify";

export default createVuetify({
  theme: {
    defaultTheme: "dark",
  },
});
