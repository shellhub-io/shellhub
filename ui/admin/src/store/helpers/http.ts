import Axios from "axios";
// eslint-disable-next-line @typescript-eslint/ban-ts-comment
// @ts-ignore //* use while route mapping is not in TypeScript
export default () => {
  const axios = Axios.create({
    baseURL: `${window.location.protocol}//${window.location.host}/admin`,
    headers: {
      Authorization: `Bearer ${localStorage.getItem("cloud_token")}`,
    },
  });

  return axios;
};
