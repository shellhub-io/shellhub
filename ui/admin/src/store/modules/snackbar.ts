import { Module } from "vuex";
import { State } from "./../index";

export interface snackbarMessageAndContentType {
  typeMessage: string;
  typeContent: string;
}

export interface SnackbarState {
  snackbarError: boolean;
  snackbarSuccess: boolean;
  snackbarMessageAndContentType: snackbarMessageAndContentType;
  snackbarCopy: boolean;
}

export const snackbar: Module<SnackbarState, State> = {
  namespaced: true,

  state: {
    snackbarError: false,
    snackbarSuccess: false,
    snackbarMessageAndContentType: { typeMessage: "", typeContent: "" },
    snackbarCopy: false,
  },

  getters: {
    snackbarSuccess: (state) => state.snackbarSuccess,
    snackbarError: (state) => state.snackbarError,
    snackbarMessageAndContentType: (state) => state.snackbarMessageAndContentType,
    snackbarCopy: (state) => state.snackbarCopy,
  },

  mutations: {
    setSnackbarSuccessAction: (state, data) => {
      state.snackbarMessageAndContentType = data;
      state.snackbarSuccess = true;
    },

    setSnackbarNoContent: (state, data) => {
      state.snackbarMessageAndContentType = data;
      state.snackbarSuccess = true;
    },

    setSnackbarSuccessDefault: (state) => {
      state.snackbarMessageAndContentType = { typeMessage: "default", typeContent: "" };
      state.snackbarSuccess = true;
    },

    unsetSnackbarSuccess: (state) => {
      state.snackbarSuccess = false;
    },

    setSnackbarErrorLoadingOrAction: (state, data) => {
      state.snackbarMessageAndContentType = data;
      state.snackbarError = true;
    },

    setSnackbarErrorCustom: (state, data) => {
      state.snackbarMessageAndContentType = data;
      state.snackbarError = true;
    },

    setSnackbarErrorDefault: (state) => {
      state.snackbarMessageAndContentType = { typeMessage: "default", typeContent: "" };
      state.snackbarError = true;
    },

    setSnackbarErrorLicense: (state, data) => {
      state.snackbarMessageAndContentType = data;
      state.snackbarError = true;
    },

    unsetSnackbarError: (state) => {
      state.snackbarError = false;
    },

    setSnackbarCopy: (state, value) => {
      state.snackbarMessageAndContentType = { typeMessage: "", typeContent: value };
      state.snackbarCopy = true;
    },

    unsetSnackbarCopy: (state) => {
      state.snackbarCopy = false;
    },
  },

  actions: {
    showSnackbarSuccessAction: ({ commit }, value) => {
      const data = { typeMessage: "action", typeContent: value };
      commit("setSnackbarSuccessAction", data);
    },

    showSnackbarSuccessDefault: ({ commit }) => {
      commit("setSnackbarSuccessDefault");
    },

    unsetShowStatusSnackbarSuccess: ({ commit }) => {
      commit("unsetSnackbarSuccess");
    },

    showSnackbarErrorLoading: ({ commit }, value) => {
      const data = { typeMessage: "loading", typeContent: value };
      commit("setSnackbarErrorLoadingOrAction", data);
    },

    showSnackbarNoContent: ({ commit }) => {
      const data = { typeMessage: "no-content", typeContent: "" };
      commit("setSnackbarNoContent", data);
    },

    showSnackbarErrorAction: ({ commit }, value) => {
      const data = { typeMessage: "action", typeContent: value };
      commit("setSnackbarErrorLoadingOrAction", data);
    },

    showSnackbarErrorCustom: ({ commit }, value) => {
      const data = { typeMessage: "custom", typeContent: value };
      commit("setSnackbarErrorCustom", data);
    },

    showSnackbarErrorLicense: ({ commit }, value) => {
      const data = { typeMessage: "licenseRequired", typeContent: value };
      commit("setSnackbarErrorLicense", data);
    },

    showSnackbarErrorDefault: ({ commit }) => {
      commit("setSnackbarErrorDefault");
    },

    unsetShowStatusSnackbarError: ({ commit }) => {
      commit("unsetSnackbarError");
    },

    showSnackbarCopy: ({ commit }, value) => {
      commit("setSnackbarCopy", value);
    },

    unsetShowStatusSnackbarCopy: ({ commit }) => {
      commit("unsetSnackbarCopy");
    },
  },
};
