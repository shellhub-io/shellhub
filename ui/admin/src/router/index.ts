import { createRouter, createWebHistory, NavigationGuardNext, RouteLocationNormalized } from "vue-router";
import Login from "@admin/views/Login.vue";
import Dashboard from "@admin/views/Dashboard.vue";
import Users from "@admin/views/Users.vue";
import SettingsLicense from "@admin/components/Settings/SettingsLicense.vue";
import SettingsAuthentication from "@admin/components/Settings/SettingsAuthentication.vue";
import Namespaces from "@admin/views/Namespaces.vue";
import Settings from "@admin/views/Settings.vue";
import { computed } from "vue";
import useLicenseStore from "@admin/store/modules/license";
import useLayoutStore from "@admin/store/modules/layout";
import useAuthStore from "@admin/store/modules/auth";
import { plugin as snackbar } from "@/plugins/snackbar"; // using direct plugin because inject() doesn't work outside components

const routes = [
  {
    path: "/login",
    name: "login",
    component: Login,
    meta: {
      layout: "SimpleLayout",
      requiresAuth: false,
    },
  },
  {
    path: "/",
    name: "dashboard",
    component: Dashboard,
  },
  {
    path: "/users",
    name: "users",
    component: Users,
  },
  {
    path: "/user/:id",
    name: "userDetails",
    component: () => import("@admin/views/UserDetails.vue"),
  },
  {
    path: "/devices",
    name: "devices",
    component: () => import("@admin/views/Device.vue"),
    redirect: {
      name: "listDevices",
    },
    children: [
      {
        path: "",
        name: "listDevices",
        component: () => import("@admin/views/Device.vue"),
      },
    ],
  },
  {
    path: "/device/:id",
    name: "deviceDetails",
    component: () => import("@admin/views/DeviceDetails.vue"),
  },
  {
    path: "/sessions",
    name: "sessions",
    component: () => import("@admin/views/Sessions.vue"),
  },
  {
    path: "/session/:id",
    name: "sessionDetails",
    component: () => import("@admin/views/SessionDetails.vue"),
  },
  {
    path: "/settings",
    name: "Settings",
    component: Settings,
    redirect: { name: "SettingProfile" },
    children: [
      {
        path: "authentication",
        name: "SettingAuthentication",
        component: SettingsAuthentication,
      },
      {
        path: "license",
        name: "SettingLicense",
        component: SettingsLicense,
      },
    ],
  },
  {
    path: "/firewall-rules",
    name: "firewall-rules",
    component: () => import("@admin/views/FirewallRules.vue"),
  },
  {
    path: "/firewall-rules/:id",
    name: "firewallRulesDetails",
    component: () => import("@admin/views/FirewallRulesDetails.vue"),
  },
  {
    path: "/namespaces",
    name: "namespaces",
    component: Namespaces,
  },
  {
    path: "/namespace/:id",
    name: "namespaceDetails",
    component: () => import("@admin/views/NamespaceDetails.vue"),
  },
  {
    path: "/announcements",
    name: "announcements",
    component: () => import("@admin/views/Announcements.vue"),
  },
  {
    path: "/announcement/:uuid",
    name: "announcementDetails",
    component: () => import("@admin/views/AnnouncementDetails.vue"),
  },
  {
    path: "/new-announcement",
    name: "new-announcement",
    component: () => import("@admin/views/NewAnnouncement.vue"),
  },
];

const router = createRouter({
  history: createWebHistory("/admin/"),
  routes,
});

router.beforeEach(
  async (to: RouteLocationNormalized, from: RouteLocationNormalized, next: NavigationGuardNext) => {
    const licenseStore = useLicenseStore();
    const layoutStore = useLayoutStore();
    const authStore = useAuthStore();

    const requiresAuth = to.meta.requiresAuth ?? true;

    const layout = to.meta.layout || "AppLayout";
    layoutStore.setLayout(layout as string);

    if (!authStore.isLoggedIn && requiresAuth) {
      return next({
        name: "login",
        query: { redirect: to.fullPath },
      });
    }

    if (authStore.isLoggedIn && !to.meta.requiresAuth) {
      const license = computed(() => licenseStore.getLicense);

      try {
        await licenseStore.get();

        if (license.value.expired && to.name !== "SettingLicense") {
          snackbar.showError("Your license has expired. Please update it and try again.");
          return next({ name: "SettingLicense" });
        }
      } catch {
        if (to.name !== "SettingLicense") {
          snackbar.showError("Failed to get your license info. Please check it and try again.");
          return next({ name: "SettingLicense" });
        }
      }
    }

    return next();
  },
);

export default router;
