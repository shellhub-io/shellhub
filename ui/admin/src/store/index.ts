import { InjectionKey } from "vue";
import { createStore, Store, useStore as vuexUseStore } from "vuex";
import { stats, StatsState } from "./modules/stats";

import { UsersState, users } from "./modules/users";
import { AuthState, auth } from "./modules/auth";
import { DevicesState, devices } from "./modules/devices";
import { firewallRules, FirewallRulesState } from "./modules/firewall_rules";
import { namespaces, NamespacesState } from "./modules/namespaces";
import { layout, LayoutState } from "./modules/layout";
import { license, LicenseState } from "./modules/license";
import { publicKeys, PublicKeysState } from "./modules/public_keys";
import { sessions, SessionsState } from "./modules/sessions";
import { snackbar, SnackbarState } from "./modules/snackbar";
import { spinner, SpinnerState } from "./modules/spinner";
import { announcement, AnnouncementState } from "./modules/announcement";
import { instance, InstanceState } from "./modules/instance";

export interface State {
  user: UsersState;
  auth: AuthState;
  devices: DevicesState;
  firewallRules: FirewallRulesState;
  layout: LayoutState;
  license: LicenseState;
  namespaces: NamespacesState;
  publicKeys: PublicKeysState;
  sessions: SessionsState;
  snackbar: SnackbarState;
  spinner: SpinnerState;
  stats: StatsState;
  announcement: AnnouncementState;
  instance: InstanceState;
}

// eslint-disable-next-line symbol-description
export const key: InjectionKey<Store<State>> = Symbol();

export const store = createStore<State>({
  modules: {
    users,
    auth,
    devices,
    firewallRules,
    layout,
    license,
    namespaces,
    publicKeys,
    sessions,
    snackbar,
    spinner,
    stats,
    announcement,
    instance,
  },
});

export function useStore(): Store<State> {
  return vuexUseStore(key);
}
