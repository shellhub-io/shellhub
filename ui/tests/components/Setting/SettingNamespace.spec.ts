import { createVuetify } from "vuetify";
import { mount, VueWrapper } from "@vue/test-utils";
import { beforeEach, describe, expect, it, vi } from "vitest";
import MockAdapter from "axios-mock-adapter";
import SettingNamespace from "@/components/Setting/SettingNamespace.vue";
import { namespacesApi, usersApi, apiKeysApi } from "@/api/http";
import { store, key } from "@/store";
import { router } from "@/router";
import { envVariables } from "@/envVariables";
import { SnackbarPlugin } from "@/plugins/snackbar";

type SettingNamespaceWrapper = VueWrapper<InstanceType<typeof SettingNamespace>>;

describe("Setting Namespace", () => {
  let wrapper: SettingNamespaceWrapper;

  const vuetify = createVuetify();

  let mockNamespace: MockAdapter;

  let mockUser: MockAdapter;

  let mockApiKeys: MockAdapter;

  const members = [
    {
      id: "507f1f77bcf86cd799439011",
      username: "test",
      role: "owner",
    },
  ];

  const namespaceData = { data: {
    name: "test",
    owner: "test",
    tenant_id: "fake-tenant",
    members,
    settings: {
      session_record: true,
      connection_announcement: "",
    },
    max_devices: 3,
    devices_count: 3,
    created_at: "",
  },
  };

  const authData = {
    status: "success",
    token: "",
    user: "test",
    name: "test",
    tenant: "fake-tenant",
    email: "test@test.com",
    id: "507f1f77bcf86cd799439011",
    role: "owner",
    mfa: {
      enable: false,
      validate: false,
    },
  };

  const session = true;

  const getKeyResponse = [
    {
      id: "3e5a5194-9dec-4a32-98db-7434c6d49df1",
      tenant_id: "fake-tenant",
      user_id: "507f1f77bcf86cd799439011",
      name: "my api key",
      expires_in: 1707958989,
    },
    {
      id: "3e5a5194-9dec-4a32-98db-7434c6d49df2",
      tenant_id: "fake-tenant",
      user_id: "507f1f77bcf86cd799439011",
      name: "my api key",
      expires_in: 1707958989,
    },
  ];

  beforeEach(async () => {
    window.matchMedia = vi.fn().mockImplementation((query) => ({
      matches: false,
      media: query,
      onchange: null,
      addListener: vi.fn(),
      removeListener: vi.fn(),
      addEventListener: vi.fn(),
      removeEventListener: vi.fn(),
      dispatchEvent: vi.fn(),
    }));

    vi.useFakeTimers();
    localStorage.setItem("tenant", "fake-tenant");
    envVariables.isCloud = true;

    mockNamespace = new MockAdapter(namespacesApi.getAxios());
    mockUser = new MockAdapter(usersApi.getAxios());
    mockApiKeys = new MockAdapter(apiKeysApi.getAxios());

    mockNamespace.onGet("http://localhost:3000/api/namespaces/fake-tenant-data").reply(200, namespaceData);
    mockUser.onGet("http://localhost:3000/api/users/security").reply(200, session);
    mockUser.onGet("http://localhost:3000/api/auth/user").reply(200, authData);
    mockUser.onGet("http://localhost:3000/api/auth/user").reply(200, authData);
    mockApiKeys.onGet("http://localhost:3000/api/namespaces/fake-tenant/api-key").reply(200, getKeyResponse, { "x-total-count": 2 });

    store.commit("auth/authSuccess", authData);
    store.commit("auth/changeData", authData);
    store.commit("namespaces/setNamespace", namespaceData);
    store.commit("security/setSecurity", session);
    store.commit("apiKeys/setKeyList", { data: getKeyResponse, headers: { "x-total-count": 2 } });

    wrapper = mount(SettingNamespace, {
      global: {
        plugins: [[store, key], vuetify, router, SnackbarPlugin],
        config: {
          errorHandler: () => { /* ignore global error handler */ },
        },
      },
    });
  });

  it("Is a Vue instance", () => {
    expect(wrapper.vm).toBeTruthy();
  });

  it("Renders the component", () => {
    expect(wrapper.html()).toMatchSnapshot();
  });

  it("Data is defined", () => {
    expect(wrapper.vm.$data).toBeDefined();
  });

  it("Data is defined", () => {
    expect(wrapper.vm.$data).toBeDefined();
  });

  const dataTests = [
    "card",
    "card-header",
    "card-title",
    "card-subtitle",
    "edit-namespace-btn",
    "profile-details-list",
    "profile-details-item",
    "name-icon",
    "name-title",
    "name-input",
    "tenant-details-item",
    "tenant-icon",
    "tenant-title",
    "tenant-copy-btn",
    "announcement-item",
    "announcement-icon",
    "announcement-title",
    "announcement-subtitle",
    "edit-announcement-btn",
    "record-item",
    "record-icon",
    "record-title",
    "record-description",
    "security-setting-component",
    "delete-leave-item",
    "delete-leave-icon",
    "delete-leave-title",
    "leave-description",
    "leave-namespace-btn",
  ];

  dataTests.forEach((dataTest) => {
    it(`should render the element with data-test="${dataTest}"`, () => {
      const element = wrapper.find(`[data-test="${dataTest}"]`);
      expect(element.exists()).toBe(true);
    });
  });
});
