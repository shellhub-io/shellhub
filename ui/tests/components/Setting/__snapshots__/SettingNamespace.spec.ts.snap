// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Setting Namespace > Renders the component 1`] = `
"<div data-v-80a16399="" class="v-container v-container--fluid v-locale--is-ltr">
  <!---->
  <!---->
  <!---->
  <!---->
  <!---->
  <!---->
  <div data-v-80a16399="" class="v-card v-theme--light v-card--density-default v-card--variant-flat bg-transparent" data-test="card">
    <!---->
    <div class="v-card__loader">
      <div class="v-progress-linear v-theme--light v-locale--is-ltr" style="top: 0px; height: 0px; --v-progress-linear-height: 2px;" role="progressbar" aria-hidden="true" aria-valuemin="0" aria-valuemax="100">
        <!---->
        <div class="v-progress-linear__background"></div>
        <div class="v-progress-linear__buffer" style="width: 0%;"></div>
        <transition-stub name="fade-transition" appear="false" persisted="false" css="true">
          <div class="v-progress-linear__indeterminate">
            <div class="v-progress-linear__indeterminate long"></div>
            <div class="v-progress-linear__indeterminate short"></div>
          </div>
        </transition-stub>
        <!---->
      </div>
    </div>
    <!---->
    <!---->
    <div data-v-80a16399="" class="v-card-item">
      <!---->
      <div class="v-card-item__content">
        <!---->
        <!---->
        <div data-v-80a16399="" class="v-list-item v-theme--light v-list-item--density-default rounded-0 v-list-item--variant-text pa-0" data-test="card-header">
          <!----><span class="v-list-item__underlay"></span>
          <!---->
          <div class="v-list-item__content" data-no-activator="">
            <div class="v-list-item-title">
              <h1 data-v-80a16399="" data-test="card-title">Namespace</h1>
            </div>
            <div class="v-list-item-subtitle"><span data-v-80a16399="" data-test="card-subtitle">Manage the namespace settings</span></div>
            <!---->
          </div>
          <div class="v-list-item__append">
            <div data-v-80a16399="" class="mr-4"><button data-v-80a16399="" type="button" class="v-btn v-theme--light text-primary v-btn--density-default v-btn--size-default v-btn--variant-text bg-secondary border" data-test="edit-namespace-btn"><span class="v-btn__overlay"></span><span class="v-btn__underlay"></span>
                <!----><span class="v-btn__content" data-no-activator="">Edit Namespace</span>
                <!---->
                <!---->
              </button></div>
            <div class="v-list-item__spacer"></div>
          </div>
        </div>
      </div>
      <!---->
    </div>
    <div data-v-80a16399="" class="v-card-text pt-4">
      <div data-v-80a16399="" class="v-list v-theme--light v-list--border v-list--density-default v-list--one-line v-list--rounded bg-background pa-0" tabindex="0" role="listbox" data-test="profile-details-list">
        <div data-v-80a16399="" class="v-card-item" style="grid-template-columns: max-content 1.5fr 2fr;" data-test="profile-details-item">
          <div class="v-card-item__prepend"><i data-v-80a16399="" class="mdi-cloud-braces mdi v-icon notranslate v-theme--light v-icon--size-default" aria-hidden="true" data-test="name-icon"></i></div>
          <div class="v-card-item__content">
            <div class="v-card-title"><span data-v-80a16399="" class="text-subtitle-1" data-test="name-title">Name</span></div>
            <!---->
            <!---->
          </div>
          <div class="v-card-item__append">
            <div data-v-80a16399="" class="v-input v-input--horizontal v-input--density-compact v-theme--light v-locale--is-ltr v-input--dirty v-input--disabled v-input--readonly v-text-field v-input--plain-underlined" data-test="name-input">
              <!---->
              <div class="v-input__control">
                <div class="v-field v-field--active v-field--disabled v-field--dirty v-field--no-label v-field--variant-plain v-theme--light v-locale--is-ltr">
                  <div class="v-field__overlay"></div>
                  <div class="v-field__loader">
                    <div class="v-progress-linear v-theme--light v-locale--is-ltr" style="top: 0px; height: 0px; --v-progress-linear-height: 2px;" role="progressbar" aria-hidden="true" aria-valuemin="0" aria-valuemax="100">
                      <!---->
                      <div class="v-progress-linear__background"></div>
                      <div class="v-progress-linear__buffer" style="width: 0%;"></div>
                      <transition-stub name="fade-transition" appear="false" persisted="false" css="true">
                        <div class="v-progress-linear__indeterminate">
                          <div class="v-progress-linear__indeterminate long"></div>
                          <div class="v-progress-linear__indeterminate short"></div>
                        </div>
                      </transition-stub>
                      <!---->
                    </div>
                  </div>
                  <!---->
                  <div class="v-field__field" data-no-activator="">
                    <!---->
                    <!---->
                    <!----><input readonly="" disabled="" size="1" type="text" id="input-v-1" aria-describedby="input-v-1-messages" required="" class="v-field__input" value="test">
                    <!---->
                  </div>
                  <!---->
                  <!---->
                  <div class="v-field__outline">
                    <!---->
                    <!---->
                  </div>
                </div>
              </div>
              <!---->
              <!---->
            </div>
          </div>
        </div>
        <hr data-v-80a16399="" class="v-divider v-theme--light" aria-orientation="horizontal" role="separator">
        <div data-v-80a16399="" class="v-card-item" style="grid-template-columns: max-content 1.5fr 2fr;" data-test="tenant-details-item">
          <div class="v-card-item__prepend"><i data-v-80a16399="" class="mdi-identifier mdi v-icon notranslate v-theme--light v-icon--size-default" aria-hidden="true" data-test="tenant-icon"></i></div>
          <div class="v-card-item__content">
            <div class="v-card-title"><span data-v-80a16399="" class="text-subtitle-1" data-test="tenant-title">Tenant ID</span></div>
            <!---->
            <!---->
          </div>
          <div class="v-card-item__append"><span data-v-80a16399="" class="v-chip v-theme--light v-chip--density-default v-chip--size-default v-chip--variant-tonal ml-1" draggable="false"><!----><span class="v-chip__underlay"></span>
            <!---->
            <!---->
            <div class="v-chip__content" data-no-activator=""><span data-v-80a16399="" aria-describedby="v-tooltip-v-4" class="hover-text" data-test="tenant-copy-btn">fake-tenant <i data-v-80a16399="" class="mdi-content-copy mdi v-icon notranslate v-theme--light v-icon--size-default" aria-hidden="true"></i></span>
              <!--teleport start-->
              <!--teleport end-->
            </div>
            <!---->
            <!----></span>
          </div>
        </div>
        <hr data-v-80a16399="" class="v-divider v-theme--light" aria-orientation="horizontal" role="separator">
        <div data-v-80a16399="" class="v-card-item" style="grid-template-columns: max-content 1.5fr 2fr;" data-test="announcement-item">
          <!---->
          <div class="v-card-item__content">
            <div class="v-card-title"><i data-v-80a16399="" class="mdi-bullhorn-variant-outline mdi v-icon notranslate v-theme--light pl-1 mr-3" style="font-size: 18px; height: 18px; width: 18px;" aria-hidden="true" data-test="announcement-icon"></i><span data-v-80a16399="" class="text-subtitle-1" data-test="announcement-title">Connection Announcement</span></div>
            <!---->
            <div data-v-80a16399="" class="v-card-text pt-1 pl-0"><span data-v-80a16399="" data-test="announcement-subtitle">A connection announcement is a custom message written during a session when a connection is established on a device within the namespace.</span></div>
          </div>
          <div class="v-card-item__append"><button data-v-80a16399="" type="button" class="v-btn v-theme--light text-aprimary v-btn--density-default v-btn--size-default v-btn--variant-text ml-4" data-test="edit-announcement-btn"><span class="v-btn__overlay"></span><span class="v-btn__underlay"></span>
              <!----><span class="v-btn__content" data-no-activator="">Edit Announcement </span>
              <!---->
              <!---->
            </button></div>
        </div>
        <hr data-v-80a16399="" class="v-divider v-theme--light" aria-orientation="horizontal" role="separator">
        <div data-v-80a16399="" class="v-row ma-0">
          <div data-v-80a16399="" class="v-card v-card--flat v-theme--light v-card--density-default v-card--variant-elevated bg-background" data-test="record-item">
            <!---->
            <div class="v-card__loader">
              <div class="v-progress-linear v-theme--light v-locale--is-ltr" style="top: 0px; height: 0px; --v-progress-linear-height: 2px;" role="progressbar" aria-hidden="true" aria-valuemin="0" aria-valuemax="100">
                <!---->
                <div class="v-progress-linear__background"></div>
                <div class="v-progress-linear__buffer" style="width: 0%;"></div>
                <transition-stub name="fade-transition" appear="false" persisted="false" css="true">
                  <div class="v-progress-linear__indeterminate">
                    <div class="v-progress-linear__indeterminate long"></div>
                    <div class="v-progress-linear__indeterminate short"></div>
                  </div>
                </transition-stub>
                <!---->
              </div>
            </div>
            <div class="v-card-item">
              <!---->
              <div class="v-card-item__content">
                <div class="v-card-title"><i data-v-80a16399="" class="mdi-play-box-outline mdi v-icon notranslate v-theme--light pl-1 mr-3" style="font-size: 18px; height: 18px; width: 18px;" aria-hidden="true" data-test="record-icon"></i><span data-v-80a16399="" class="text-subtitle-1" data-test="record-title">Session Record</span>
                  <div data-v-80a16399="" class="v-card-text pl-0 pt-1" data-test="record-description"> Session record is a feature that allows you to check logged activity when connecting to a device. </div>
                </div>
                <!---->
                <!---->
              </div>
              <!---->
            </div>
            <!---->
            <!---->
            <!---->
            <!----><span class="v-card__underlay"></span>
          </div>
          <div data-v-80a16399="" class="v-col d-flex align-center justify-end bg-background">
            <div data-v-80a16399="" class="v-input v-input--horizontal v-input--center-affix v-input--density-default v-theme--light v-locale--is-ltr v-input--dirty v-switch v-switch--inset" data-test="session-recording-setting-component">
              <!---->
              <div class="v-input__control">
                <div class="v-selection-control v-selection-control--dirty v-selection-control--density-default">
                  <div class="v-selection-control__wrapper text-primary">
                    <div class="v-switch__track bg-primary">
                      <!---->
                      <!---->
                    </div>
                    <div class="v-selection-control__input"><input checked="" id="switch-v-6" aria-disabled="false" type="checkbox" aria-describedby="switch-v-6-messages" value="true">
                      <div class="v-switch__thumb">
                        <transition-stub name="scale-transition" appear="false" persisted="false" css="true">
                          <!---->
                        </transition-stub>
                      </div>
                    </div>
                  </div>
                  <!---->
                </div>
              </div>
              <!---->
              <!---->
            </div>
          </div>
        </div>
        <hr data-v-80a16399="" class="v-divider v-theme--light" aria-orientation="horizontal" role="separator">
        <div data-v-80a16399="" class="v-card-item" style="grid-template-columns: max-content 1.5fr 2fr;" data-test="delete-leave-item">
          <!---->
          <div class="v-card-item__content">
            <div class="v-card-title"><i data-v-80a16399="" class="mdi-delete mdi v-icon notranslate v-theme--light pl-1 mr-3" style="font-size: 18px; height: 18px; width: 18px;" aria-hidden="true" data-test="delete-leave-icon"></i><span data-v-80a16399="" class="text-subtitle-1" data-test="delete-leave-title">Leave Namespace</span></div>
            <!---->
            <div data-v-80a16399="" class="v-card-text pt-1 pl-0"><span data-v-80a16399="" data-test="leave-description">After leaving a namespace, you will need to be invited again to access it.</span></div>
          </div>
          <div class="v-card-item__append"><button data-v-80a16399="" type="button" class="v-btn v-theme--light text-error v-btn--density-default v-btn--size-default v-btn--variant-text" data-test="leave-namespace-btn"><span class="v-btn__overlay"></span><span class="v-btn__underlay"></span>
              <!----><span class="v-btn__content" data-no-activator="">Leave</span>
              <!---->
              <!---->
            </button></div>
        </div>
      </div>
    </div>
    <!---->
    <!----><span class="v-card__underlay"></span>
  </div>
</div>"
`;
